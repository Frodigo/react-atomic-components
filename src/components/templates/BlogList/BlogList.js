import React from 'react';

import PageHeader from '../../organisms/PageHeader/PageHeader';
import Tile from '../../molecues/Tile';
import Breadcrumbs from '../../molecues/Breadcrumbs';
import ContentBlock from '../../molecues/ContentBlock';
import BlogArticlesList from '../../organisms/BlogArticlesList';
import Newsletter from '../../organisms/Newsletter';
import PageFooter from '../../organisms/PageFooter/PageFooter';
import MobileNav from '../../organisms/MobileNav/MobileNav';
import { items as mobileNavItems } from '../../organisms/MobileNav/MobileNav.config';
import FooterMock from "../partials/FooterMock";

const BlogList = () => {
    const breadcrumbsItems = [
        {
            label: 'Home',
            url: '#'
        },
        {
            label: 'Blog',
            url: '#'
        },
        {
            label: 'PWA Studio',
            url: '#'
        }
    ];

    const newsletterProps = {
        title: 'Subscribe to the newsletter',
        subtitle: 'I will send you unique content every Friday.',
        submitUrl: 'https://app.getresponse.com/add_subscriber.html',
        token: 'WMdq9',
        thankYouPageUrl: 'https://marcin-kwiatkowski.com/thank-you',
        footer:
            'Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers',
        onSubmit: (data, e) => e.target.submit()
    };

    const articles = [
        {
            id: 1,
            title: 'Lorem ipsum',
            author: 'Andrew',
            date: '6 November 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. '
        },
        {
            id: 2,
            title: 'Lorem ipsum',
            author: 'Andrew',
            date: '9 October 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. '
        },
        {
            id: 3,
            title: 'Lorem ipsum',
            author: 'Andrew',
            date: '25 September 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. '
        },
        {
            id: 4,
            title: 'Lorem ipsum',
            author: 'Andrew',
            date: '31 August 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. '
        },
        {
            id: 5,
            title: 'Lorem ipsum',
            author: 'Andrew',
            date: '24 August 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. '
        },
        {
            id: 6,
            title: 'Lorem ipsum',
            author: 'Andrew',
            date: '15 August 2020',
            url: '#',
            image: 'https://via.placeholder.com/396x222',
            shortDescription:
                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. '
        }
    ];

    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home',
            isActive: true
        },
        {
            url: '#',
            label: 'About me'
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ];

    return (
        <>
            <PageHeader navItems={pageHeaderNavItems} />
            <Tile heading="PWA Studio" />
            <Breadcrumbs items={breadcrumbsItems} />
            <ContentBlock
                heading="Lorem ipsum"
                paragraph="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi blandit tortor risus, nec finibus velit eleifend sit amet. Morbi ullamcorper pellentesque ante quis aliquet. Vestibulum convallis dolor et neque congue fermentum. "
                cta={[{
                    href: '#',
                    children: 'Read more',
                    variant: 'primary'
                }]}
            />
            <BlogArticlesList items={articles} />
            <FooterMock/>
        </>
    );
};

export default BlogList;
