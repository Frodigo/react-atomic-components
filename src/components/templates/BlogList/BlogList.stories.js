import React from 'react';

import { default as BlogListComponent } from './BlogList';

export default {
    title: 'Templates',
    component: BlogListComponent
};

const Template = (args) => <BlogListComponent {...args} />;

export const BlogList = Template.bind({});
