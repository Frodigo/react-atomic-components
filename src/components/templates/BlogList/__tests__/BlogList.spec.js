import React from 'react';
import { render } from '@testing-library/react';

import BlogList from '../BlogList';

describe('Blog list should be rendered', () => {
    test('Blog list renders correctly', () => {
        const { container } = render(<BlogList />);

        expect(container).toMatchSnapshot();
    });
});
