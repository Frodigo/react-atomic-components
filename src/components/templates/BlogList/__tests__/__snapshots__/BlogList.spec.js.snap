// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog list should be rendered Blog list renders correctly 1`] = `
<div>
  <header
    class="pageHeader"
  >
    <div
      class="container"
    >
      <a
        class="link"
        href="/"
      >
        <img />
      </a>
      <nav
        class="primaryNav"
      >
        <a
          class="navLinkActive navLink"
          href="#"
        >
          Home
        </a>
        <a
          class=" navLink"
          href="#"
        >
          About me
        </a>
        <a
          class=" navLink"
          href="#"
        >
          Blog
        </a>
        <a
          class=" navLink"
          href="#"
        >
          Get in touch
        </a>
      </nav>
    </div>
  </header>
  <section
    class="tile"
  >
    <h2
      class="heading"
    >
      PWA Studio
    </h2>
  </section>
  <nav
    class="breadcrumbs"
    items="[object Object],[object Object],[object Object]"
  >
    <div
      class="container"
    >
      <a
        class="breadcrumbsItem"
        href="#"
      >
        <span
          class="label"
        >
          Home
        </span>
      </a>
      <a
        class="breadcrumbsItem"
        href="#"
      >
        <span
          class="label"
        >
          Blog
        </span>
      </a>
      <span
        class="activeItem"
      >
        PWA Studio
      </span>
    </div>
  </nav>
  <section
    class="block"
    style="background-color: transparent;"
  >
    <div
      class="container"
    >
      <h3
        class="heading"
      >
        Lorem ipsum
      </h3>
      <p
        class="paragraph"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi blandit tortor risus, nec finibus velit eleifend sit amet. Morbi ullamcorper pellentesque ante quis aliquet. Vestibulum convallis dolor et neque congue fermentum. 
      </p>
      <div
        class="links"
      >
        <a
          class="primary"
          href="#"
        >
          Read more
        </a>
      </div>
    </div>
  </section>
  <div
    class="list"
  >
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Lorem ipsum
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Andrew
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1604617200000"
              >
                6 November 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. 
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Lorem ipsum"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Lorem ipsum
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Andrew
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1602194400000"
              >
                9 October 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. 
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Lorem ipsum"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Lorem ipsum
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Andrew
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1600984800000"
              >
                25 September 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. 
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Lorem ipsum"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Lorem ipsum
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Andrew
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1598824800000"
              >
                31 August 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. 
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Lorem ipsum"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Lorem ipsum
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Andrew
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1598220000000"
              >
                24 August 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. 
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Lorem ipsum"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
    <div
      class="block"
    >
      <div
        class="blockInner"
      >
        <section
          class="text"
        >
          <h2
            class="heading"
          >
            <a
              href="#"
            >
              Lorem ipsum
            </a>
          </h2>
          <div
            class="meta"
          >
            <span
              class="author"
            >
              Andrew
            </span>
            <span
              class="metaItem"
            >
              <time
                datetime="1597442400000"
              >
                15 August 2020
              </time>
            </span>
          </div>
          <a
            class="primary"
            href="#"
          >
            Read more
          </a>
        </section>
        <div
          class="shortDescription"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque condimentum justo et arcu auctor, nec semper tellus dignissim. Integer iaculis arcu dui, vel maximus ante faucibus sed. Fusce tempor tincidunt neque, euismod lacinia mauris pharetra a. Proin fermentum orci ex, eget laoreet nunc luctus eget. Curabitur ut leo ac justo tincidunt ornare nec eu lacus. Donec eleifend dapibus neque in ornare. 
             
            <a
              href="#"
            >
              Read more
            </a>
          </p>
        </div>
        <figure
          class="figure"
        >
          <a
            href="#"
          >
            <img
              alt="Lorem ipsum"
              src="https://via.placeholder.com/396x222"
              width="500"
            />
          </a>
        </figure>
      </div>
    </div>
  </div>
  <form
    action="https://app.getresponse.com/add_subscriber.html"
    class="newsletter"
    id="newsletter"
    method="post"
  >
    <div
      class="newsletterContent"
    >
      <h3
        class="title"
      >
        Subscribe to the newsletter
      </h3>
      <p
        class="paraghaph"
      >
        I will send you unique content every Friday.
      </p>
      <div
        class="fieldset"
      >
        <div
          class="field"
        >
          <input
            class="input null"
            data-testid="input"
            name="first_name"
            placeholder="Your name"
            type="text"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <input
            class="input null"
            data-testid="input"
            name="email"
            placeholder="Your email"
            type="email"
            value=""
          />
        </div>
      </div>
      <input
        name="campaign_token"
        type="hidden"
        value="WMdq9"
      />
      <input
        name="thankyou_url"
        type="hidden"
        value="https://marcin-kwiatkowski.com/thank-you"
      />
      <input
        name="start_day"
        type="hidden"
        value="0"
      />
      <footer
        class="footer"
      >
        <p>
          Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers
        </p>
        <button
          class="primary"
          type="submit"
        >
          Subscribe
        </button>
      </footer>
    </div>
  </form>
  <footer
    class="pageFooter"
  >
    <div
      class="columns"
    >
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 1
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 2
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 3
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 4
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
    </div>
    <div>
      <ul
        class="bottomNav"
      >
        <li>
          <a
            href="#"
          >
            Link 1
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Link 2
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Link 3
          </a>
        </li>
      </ul>
      <p
        class="copyright"
      >
        © 2021 Lorem ipsum
      </p>
    </div>
  </footer>
  <nav
    class="mobileNav"
  >
    <a
      href="/"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Home"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Home
        </span>
      </span>
    </a>
    <a
      href="/blog"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Blog"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Blog
        </span>
      </span>
    </a>
    <a
      href="/about-me"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="About"
          src="[object Object]"
        />
        <span
          class="label"
        >
          About
        </span>
      </span>
    </a>
    <a
      href="/get-in-touch"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Contact"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Contact
        </span>
      </span>
    </a>
    <a
      href="#newsletter"
    >
      <span
        class="iconContainer"
      >
        <img
          alt="Subscribe"
          src="[object Object]"
        />
        <span
          class="label"
        >
          Subscribe
        </span>
      </span>
    </a>
  </nav>
</div>
`;
